frontend/src/Components/Endpoint/Endpoint.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import Button from "plaid-threads/Button";
+import Note from "plaid-threads/Note";
+
+import Table from "../Table/Table";
+import Error from "../Error/Error";
+
+import {
+  DataItem,
+  Categories,
+  ErrorDataItem,
+  Data,
+} from "../../Utilities/dataUtilities";
+
+import styles from "./Endpoint.module.scss";
+
+interface Props {
+  endpoint: string;
+  name?: string | null;
+  categories: Array<Categories>;
+  schema: string;
+  description: string;
+  transformData: (arg: any) => Array<DataItem>;
+}
+
+const Endpoint = (props: Props) => {
+  const [showTable, setShowTable] = useState(false);
+  const [transformedData, setTransformedData] = useState<Data>([]);
+  const [isError, setIsError] = useState(false);
+  const [error, setError] = useState<ErrorDataItem>({
+    error_type: "",
+    error_code: "",
+    error_message: "",
+    display_message: "",
+    status_code: null,
+  });
```

Comment:
could the type of this be `ErrorDataItem | null` and just initialize it to `null`? Then we dont need a separate `isError` state, we can just check `error != null`

Reply:
done